enterprise/frontend/src/metabase-enterprise/google_drive/utils.tsx

```diff
@@ -1,25 +1,78 @@
+import { P, match } from "ts-pattern";
+import { t } from "ttag";
+
 import { skipToken } from "metabase/api";
-import { useSetting } from "metabase/common/hooks";
+import { useHasTokenFeature, useSetting } from "metabase/common/hooks";
 import { useSelector } from "metabase/lib/redux";
 import { getUserIsAdmin } from "metabase/selectors/user";
 import { useGetServiceAccountQuery } from "metabase-enterprise/api";
+import type { GdrivePayload } from "metabase-types/api";
 
 export function useShowGdrive() {
-  const gSheetsSetting = useSetting("gsheets");
   const gSheetsEnabled = useSetting("show-google-sheets-integration");
+  const hasDwh = useHasTokenFeature("attached_dwh");
   const userIsAdmin = useSelector(getUserIsAdmin);
 
-  const shouldGetServiceAccount = gSheetsEnabled && userIsAdmin;
+  const shouldGetServiceAccount = gSheetsEnabled && userIsAdmin && hasDwh;
 
-  const { data: { email: serviceAccountEmail } = {} } =
-    useGetServiceAccountQuery(shouldGetServiceAccount ? undefined : skipToken);
+  const { data: serviceAccount } = useGetServiceAccountQuery(
+    shouldGetServiceAccount ? undefined : skipToken,
+  );
 
   const showGdrive = Boolean(
-    gSheetsEnabled &&
-      gSheetsSetting?.status &&
-      userIsAdmin &&
-      serviceAccountEmail,
+    hasDwh && gSheetsEnabled && userIsAdmin && serviceAccount?.email,
   );
 
   return showGdrive;
 }
+
+export const getStatus = ({
+  status,
+  error,
+}: {
+  status: GdrivePayload["status"] | undefined | null;
+  error?: unknown | null;
+}): GdrivePayload["status"] =>
+  match({ error: !!error, status })
+    .returnType<GdrivePayload["status"]>()
+    .with({ error: true }, () => "error")
+    .with({ status: P.string.minLength(1) }, ({ status }) => status)
+    .otherwise(() => "not-connected");
+
+export const getErrorMessage = (
+  payload:
+    | unknown
+    | string
+    | { data: { message: string } | string }
+    | { message: string },
+  fallback: string = t`Something went wrong`,
+): string => {
+  if (!payload || typeof payload !== "object") {
+    return fallback;
+  }
+
+  if (typeof payload === "string") {
+    return payload;
+  }
```

Comment:
I don't think this case will ever fire because of `typeof payload !== "object"` on line 50.

Reply:
I know this, but typescript does not!

Reply:
good catch, I think I reordered those blocks stupidly